hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```
@@ -3713,6 +3739,12 @@ S3AFileStatus s3GetFileStatus(final Path path,
         }
       } catch (AmazonClientException e) {
         throw translateException("getFileStatus", path, e);
+      }  catch (AwsServiceException e) {
+        // TODO: This and above exception handling blocks will be updated during exception
+        //  handling work.
+        if (e.statusCode() != SC_404 || isUnknownBucketV2(e)) {
+          throw translateExceptionV2("getFileStatus", path.toString(), e);
+        }
       }
```

I saw this was related to FNFE but don't really know from comment why exactly we're just doing nothing here. why not just let it throw?